DuggaSys/tests/courseedservice-test.php

```diff
@@ -0,0 +1,254 @@
+/*
+----------------------------------------------------------
+    Courseedservice.php implement test file
+----------------------------------------------------------
+*/
+ 
+<?php
+ 
+include "../../Shared/test.php";
+ 
+$testsData = array(
+
+    /* Insert into course */
+
+    'Insert into course test' => array(
+        'expected-output' => '{"debug":"NONE!","motd":"UNK"}',
+        'query-before-test-1' => "SELECT cid FROM course WHERE coursecode = 'G1337' ORDER BY coursecode DESC LIMIT 1",
+        'query-before-test-2' => "INSERT INTO course (coursecode,coursename,visibility) VALUES('G1337','Testing-Course',1)",
+        'query-before-test-3' => "DELETE FROM course WHERE coursecode = 'G1337' AND cid = ?",
+        'query-after-test-1' => "DELETE FROM course WHERE coursecode = 'G1337' AND coursename = 'Testing-Course'",
+        'query-after-test-2' => "DELETE FROM course WHERE coursecode = 'G1337' AND coursename = 'Testing-Course'",
+	'query-variables' => "blop",
+        'service' => 'https://cms.webug.se/root/G2/students/a21ponkr/LenaSYS/DuggaSys/courseedservice.php',
+        'service-data' => serialize(array(
+            'opt' => 'NEW',
+            'cid' => '1885',
+            'username' => 'usr',
+            'password' => 'pass',
+            'coursecode' => 'G1337',
+            'coursename' => 'Testing-Course',
+            'activeversion' => '1337',
+            'activeedversion' => null,
+            'registered' => false,
+	    'blop' => '<!query-before-test-1!> <*[0][cid]*>'
+        )),
+        'filter-output' => serialize(array(
```

Comment:
As mentioned before, some data might need to be manually removed from the JSON string. The data that is to be removed is decided here, or rather you specify what values are to be compared, these values will then be gathered from the service. But they will not automatically get filtered in the expected output, which is why you need to do it manually.

That said, you need to specify what these values are. 
When you have performed the action on LenaSYS that corresponds with the service that is to be tested, you take the output (same output as the expected output), and past it into this subservience. 

https://jsonformatter.org/json-pretty-print
if you change the display option to form, you will see what values are needed.
![image](https://github.com/HGustavs/LenaSYS/assets/102600391/2890e463-f649-45ba-90e1-a3912458ee46)

here you have an example from my own test. 
![image](https://github.com/HGustavs/LenaSYS/assets/102600391/8920319f-d13a-43d2-b20f-596e0c21139a)
notice that you can specify values inside of the arrays as well if need be.